Classes/Domain/Model/AttributeSet.php

```diff
@@ -132,6 +135,18 @@ public function getAttributes(): ObjectStorage
         return $this->attributes;
     }
 
+    /**
+     * Return attributes included in listings.
+     *
+     * @return array
+     */
+    public function getListingAttributes(): array
+    {
+        return $this->collection($this->getAttributes())
+            ->filter(fn (Attribute $attribute) => $attribute->isShowInAttributeListing())
```

Comment:
Wouldn't it be better to just have `showInAttributeListing()`?

```suggestion
            ->filter(fn (Attribute $attribute) => $attribute->showInAttributeListing())
```